src/components/DurationMenu.js

```diff
@@ -0,0 +1,49 @@
+import { useState } from 'react';
+
+function DurationMenu (props) {
+
+    // For making a visable focus since the radio is hidden
+    const [focusOn, setFocusOn] = useState('');
+    const onFocus = (e) => {
+        setFocusOn(`${e.target.value}`)
+    }
+
+    // Remove focus styling from radios when focus moves to button
+    const onBlur = () => {
+        setFocusOn('');
+    }
+
+    //Send selected value up to calling component
+    const onClick = (e) => {
+        props.onSelect(e.target.attributes.for.value);
+    }
+
+    const durationToTimeSpan = (duration) => {
+        return duration === 'Annual' ? 'year' : 'month';
+    }
+
+    return (
+        <div className='flex flex-col items-center xxs:flex-row m-4'>
+            {props.options.map(option => {
+                return (
+                    <div key={option.duration} className={`inline m-1 p-4 pb-0 w-full xxs:w-1/2 bg-very-pale-blue rounded-xl ${focusOn === option.duration && 'outline outline-bright-blue'}`}>
+
+                        <label htmlFor={option.duration}
+                        onClick={onClick}
+                        className='block cursor-pointer underline text-desaturated-blue hover:text-bright-blue/80 hover:font-extrabold font-bold first-line:text-bright-blue first-line:underline'>
+                            {/* Content of the label */}
+                            {`${option.duration}`}
+                            <br />
+                            {`$${option.price}/${durationToTimeSpan(option.duration)}`}
+                        </label>
+
+                        {/* Hidden radio input */}
+                        <input type='radio' id={option.duration} name='duration' value={option.duration} onFocus={onFocus} onBlur={onBlur} className='opacity-0' />
```

Comment:
Having a `onFocus` and `onBlur` functions to handle the focus states is quite an interesting approach here. If you wanted to simplify how this works, there are couple of ways we could achieve this in CSS.

One approach could be to remove the `opacity-0` from the input, and style it so that it fills the container (e.g. absolute positioning, 100% height and width etc). It will need to have a focus state, if the default one has been removed and also `appearance-none`, so it doesn't display the native browser input styling. When a user focuses on the element, it will still look like they are focusing on the parent div, but they are actually focussing on the input that is filling the div.

An alternative approach could be to style the label so that it fills the container, and then switch the order of the `input` and `label` in the `return` (so input comes first). Then the CSS sibling selector (it looks like Tailwind uses a `peer` class https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-sibling-state) could be used to style the label, based on the whether the input is selected or not. A similar example to this can be found here https://www.rafbf.org/donate.

Obviously the current approach works just fine, but if you wanted to reduce some of the complexity then there are couple of CSS-only approaches available. I think then the state hooks could be removed too, which would offer a slight performance benefit

Reply:
I will have a play around with your suggestions, I do like CSS-only solutions.